### PR TITLE
Tracking old CLI clients in the server.

### LINKED ISSUE #464: Track Old Clients in the Server
**As a** maintainer of the Zally project
**I want** to know if there are any old clients using the linter API
**So that** the safe deprecation of old clients is assured

**Acceptance Criteria**
- Current version of the client is specified per configuration
- Requests from the older Zally CLI versions are logged in a parsable format
- Log analysis is done for the last n days

### PR DESCRIPTION
**Pull request by maxim-tschumak**
Closes #464.

### CODE REVIEW

File: server/src/main/java/de/zalando/zally/apireview/ServerMessageService.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    private String releasesPage;
+
+    ServerMessageService(@Value("#{'${zally.cli.currentCLIVersions}'.split(',')}") List<String> currentCLIVersions,
+                         @Value("${zally.cli.releasesPage:}") String releasesPage){
+        this.currentCLIVersions = currentCLIVersions;
+        this.releasesPage = releasesPage;
+    }
+
+    protected String serverMessage(String userAgent) {
+        if (userAgent == null || userAgent.isEmpty() || currentCLIVersions.contains(version(userAgent))) {
```
Comment (netme) [inline server/src/main/java/de/zalando/zally/apireview/ServerMessageService.java]: Can be false-positive, e.g. when `web-ui` is querying us. Can we just check for 1.0 and report that this client is deprecated?

File: server/src/main/resources/application-dev.yml [hunk 2]
```diff
@@ -27,3 +27,6 @@ management.port: 8080
 
 zally:
   ignoreRules: S001,S010
+  cli:
+    releasesPage: https://github.com/zalando-incubator/zally/releases
+    currentCLIVersions: 1.0,1.1
```
Comment (netme) [inline server/src/main/resources/application-dev.yml]: And then we can rename it to `deprecatedCliVersions: 1.0`

File: server/src/main/java/de/zalando/zally/apireview/ServerMessageService.java (hunk 1 above)
Comment (maxim-tschumak) [inline server/src/main/java/de/zalando/zally/apireview/ServerMessageService.java]: Do all out CLI versions send the `user-agent` header? If not - we will be not able to track those.

Comment (netme) [inline server/src/main/java/de/zalando/zally/apireview/ServerMessageService.java]: @maxim-tschumak , I guess that's something to check. My guess it's uses default user-agent from Unirest library:
https://github.com/Mashape/unirest-java/blob/3b461599ad021d0a3f14213c0dbb85bab7244f66/src/main/java/com/mashape/unirest/http/HttpClientHelper.java#L69

Could you please verify that with existing 1.0 build?

Comment (maxim-tschumak) [inline server/src/main/java/de/zalando/zally/apireview/ServerMessageService.java]: Ok, will do this!

**Comment by maxim-tschumak** [comment]
@netme can you please have another look?

**Comment by netme** [comment]
👍

**Comment by netme** [comment]
Restarted Travis, hope it will pass

**Comment by netme** [comment]
Cool, Travis is now passing. @maxim-tschumak, waiting a self-approval from you :-)

**Comment by maxim-tschumak** [comment]
:+1:

**Pull request merged**